### ISSUE TITLE
Setting default webDav server via .env file - or some other method

### ISSUE THREAD

**Issue by dmorlitz** [200ok-ch/organice#827]
I have noticed that there is a way to set default credentials for multiple storage backends through the .env file.  I was wondering if there was a way to set the default webDav server name, and possibly the username through this file - or any other method.  That would make signing into organice much easier - especially when on a mobile device.

For self-hosted environments, I imagine that the webDav server name is almost always identical.  Even without a default username, I could easily use a password manager to autofill the login/password combination.......but getting 3 fields filled in with a password manager does not always seem to work.

**Comment by munen** [comment]
Hi @dmorlitz 

Thank you for using organice 🙏🏻

You are right. It certainly can make sense for self hosted instances to have a pre-filled WebDAV host. 

Some background: The other credentials in the .env file are of a different kind, though. For Dropbox and Gitlab, we have to set up a so called “app” with either of these providers so that we can provide OAuth authentication. The current credentials in the .env file are for these apps, not for the users credentials for the particular host. 

Having said that, WebDAV does not require a more complicated auth strategy like OAuth. It would certainly be possible to use the .env file for WebDAV defaults. 

Do you want to work on that? If you prefer some help, I can do it, as well. 

Kind regards

**Comment by dmorlitz** [comment]
I am willing to give it a shot - if you pointed me in the right direction.
I am sort of a hack (and not really a Node expert) but it never hurts to
learn.

If I can figure it out, then I will submit a PR.

**Comment by munen** [comment]
It’s a good first issue to get your feet wet^^

Here’s the react hook which sets the default value for the WebDAV server in the sign in form: https://github.com/200ok-ch/organice/blob/6a5721b47bb1b4f3ef6f3d820ec2d1dc6e0bb18a/src/components/SyncServiceSignIn/index.js#L22

As you can see, currently the default is an empty string. Instead, you could read this value from the .env file. An example on how to read from this, you can find in the sync backends. I’m currently in a social setting and don’t have time to look it up. If you can’t find it, let me know and I’ll look it up later!

Good luck and have fun^^

**Comment by dmorlitz** [comment]
Thank you very much for the pointers.  I was able to easily find where the
variable was filled out - and I used the other sync backends as an example
of pulling the environment variable from the .env file.

This feature only seems to require 2 updates - which was easy to get
started with.

Now, I just have to learn how to use the git cli better, so I can exclude
the package-lock.json and package.json from my commits.  As a quick
workaround on this one, I just edited the files online in the Git interface
(it took me longer to get a "clean" commit than it did to find and update
the code.

Thank you for your help.  This is a SPECTACULAR project and I am glad that
I found it.

**Comment by dmorlitz** [comment]
Since I have submitted a PR for this issue - I am going to close this question thread.

**Comment by munen** [comment]
> Thank you for your help.  This is a SPECTACULAR project and I am glad that I found it.

Thank you for your support and your kind words :pray:  These go a long way!

We have recently created a landing page for organice and I would like to add some user testimonials. Would you mind if I use "This is a SPECTACULAR project and I am glad that I found it." with your GH username and image on there? If you do, that's perfectly alright and I will not use it in this way, of course^^

**Comment by dmorlitz** [comment]
Please feel free to include that as a testimonial.  I am really big into
the principals of open-source as well as self-hosting.  (After you do
security consulting for a while, you tend to get paranoid)  I would even
expand the testimonial to say:

"This is a SPECTACULAR project and I am glad that I found it.  Organice
takes the extreme power of org mode and makes it simple to use.  When
combined with the option to self-host your own data organice has created an
extremely flexible and secure option to help fulfill everyone's
organizational needs."

**Comment by munen** [comment]
Perfect, thank you very much :bow: 

And I wholeheartedly agree with you on the issue that it is of utmost importance to have options to make our data our own. That was initially the big motivation behind using Org mode for me. Before that, I used many other proprietary tools - and not all of them let me migrate my data. Org mode will always allow it, and with https://github.com/200ok-ch/org-parser we are trying to make it more accessible from different languages and tools, as well(;

Since I switched many years ago, my use of Org mode has only expanded, because I realized how much more can be done with it compared to what I used before (Evernote, Things, etc). So, even though morals and security were the initial drivers, UX and functionality have not suffered, they were expanded^^
